easybuild/tools/docs.py

```diff
@@ -744,31 +744,59 @@ def list_toolchains_rst(tcs):
     """ Returns overview of all toolchains in rst format """
     title = "List of known toolchains"
 
-    # figure out column names
-    table_titles = ['name', 'compiler', 'MPI']
-    for tc in tcs.values():
-        table_titles.extend(tc.keys())
+    # Specify the column names for the table
+    table_titles = ['name', 'compiler', 'MPI', 'BLAS', 'FFT']
 
+    # Set up column name : display name pairs
     col_names = {
-        'COMPILER_CUDA': 'CUDA compiler',
-        'SCALAPACK': 'ScaLAPACK',
+        'name': 'Name',
+        'compiler': 'Compiler(s)',
+        'BLAS': "BLAS/LAPACK/ScaLAPACK",
     }
 
+    # removes duplicate entries
     table_titles = nub(table_titles)
 
+    # Initialize an empty list of lists for the table data
     table_values = [[] for i in range(len(table_titles))]
+
+    # Fill in the first column with the names of the toolchains
     table_values[0] = ['**%s**' % tcname for tcname in tcs.keys()]
 
+    # Fill in the other table values
     for idx in range(1, len(table_titles)):
         for tc in tcs.values():
-            table_values[idx].append(', '.join(tc.get(table_titles[idx].upper(), [])))
+            table_values[idx].append(', '.join(tc.get(table_titles[idx].upper(), ["(*none*)"])))
 
+    # We want to combine the display for linear algebra, so create a new list
+    # with a list of BLAS, LAPACK, and ScaLAPACK included.
+    blas = []
+    for tc in tcs.values():
+        entry = list(tc.get('BLAS', ['(*none*)']))
+        entry.extend(list(tc.get('LAPACK', [])))
+        entry.extend(list(tc.get('SCALAPACK', [])))
+        blas.append(", ".join(nub(entry)))
```

Comment:
I would collapse this with the for loop above, and also handle the first "name" column in the same loop.

Also, I would first create a sorted list of toolchain names, which avoids the transposing below.

```python
# first obtained a sorted list of toolchain names
sorted_tc_names = sorted(tc.keys(), key=str.lower)

# fallback value, to avoid empty table elements
none = '*(none)*'

for col_id, col_name enumerate(table_titles):
    if col_name == 'name':
        # toolchain names column
        table_values[col_id] = ['**%s**' % tcname for tcname in sorted_tc_names]
    else:
        for tc_name in sorted_tc_names:
            tc = tcs[tc_name]
            if col_name == 'linalg':
                # column for linear algebra toolchain elements is combination of BLAS/LAPACK/ScaLAPACK values
                entry = ', '.join(tc.get(key) or none for key in ['BLAS', 'LAPACK', 'SCALAPACK'])
            else:
                # for other columns, we can grab the values via 'tc', using the upper case column name as key
                key = col_name.upper()
                entry = ', '.join(tc.get(key)) or none

            table_values[col_id].append(entry)

Reply:
This took a bit of massaging, but thanks for the framework.  Hopefully the new version is OK.